app/models/profile.rb

```diff
@@ -51,7 +61,16 @@ def self.mapped_attributes
   # the profiles table doesn't exist yet (e.g. when running bin/setup in a new
   # clone). I wish Rails had a hook for code to run after the app started, but
   # for now this is the best I can come up with.
-  refresh_attributes! if ApplicationRecord.connection.table_exists?("profiles")
+  def self.db_ready?
+    ActiveRecord::Base.connection
+  rescue ActiveRecord::NoDatabaseError
+    false
+  else
+    ActiveRecord::Base.connection.table_exists?("profiles")
+  end
+  private_class_method :db_ready?
+
+  refresh_attributes!
```

Comment:
Ruby evaluates class bodies as it reads the class, so you can actually call methods in them (that's how things like `belongs_to` work). Here we call `refresh_attributes!` to set up all the store accessors for the currently defined profile fields.